app/components/TaskList.tsx

```diff
@@ -30,6 +59,50 @@ const TaskList: React.FC<Props> = ({ projectRevision }) => {
 
   let mode = projectByProjectId ? "update" : "create";
 
+  function getStatus(forms: any[]) {
+    let status = "Not Started";
+    for (const form of forms) {
+      if (form?.changeStatus === "pending" && !form.isPristine) {
+        status = "Incomplete";
+        break;
+      }
+      if (
+        form?.changeStatus === "staged" &&
+        form?.validationErrors.length > 0
+      ) {
+        status = "Attention Required";
+        break;
+      }
+      if (
+        form?.changeStatus === "staged" &&
+        form?.validationErrors.length === 0
+      ) {
+        status = "Complete";
+      }
+    }
+    return status;
+  }
+  const projectForms = [projectFormChange];
+  const projectContactForms = tasklistProjectContactFormChanges.edges.map(
+    ({ node }) => node
+  );
+  const projectManagerForms = projectManagerFormChangesByLabel.edges
+    .filter(({ node }) => node)
+    .map(({ node }) => node.formChange);
+
+  const projectOverviewStatus = useMemo(
+    () => getStatus(projectForms),
+    projectForms
+  );
+  const projectManagerStatus = useMemo(
+    () => getStatus(projectManagerForms),
+    projectManagerForms
+  );
+  const projectContactStatus = useMemo(
+    () => getStatus(projectContactForms),
+    projectForms
+  );
+
```

Comment:
The code above won't do anything useful with `useMemo` as `projectForms`, `projectContactForms` and `projectManagerForms` will be new objects on every render. There's also a mistake with the dependency for `projectContactStatus` being `projectForms` instead of `projectContactForms`, ESlint should give you a warning about that, no?

something like this, with better formatting, is what you need here:

```suggestion
  const projectOverviewStatus = useMemo(
    () => getStatus([projectFormChange]),
    projectFormChange
  );
  const projectManagerStatus = useMemo(
    () => getStatus(projectManagerFormChangesByLabel.edges
    .filter(({ node }) => node)
    .map(({ node }) => node.formChange)),
    projectManagerFormChangesByLabel
  );
  const projectContactStatus = useMemo(
    () => getStatus(tasklistProjectContactFormChanges.edges.map(
    ({ node }) => node
  )),
    tasklistProjectContactFormChanges
  );

```